pkg/input/arm.go

```diff
@@ -59,17 +59,43 @@ func (c *ArmDetector) DetectFile(i *File, opts DetectOptions) (IACConfiguration,
 		discovered[d.name.String()] = d
 	}
 
+	processedVariables, err := processVariables(template.Variables)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", FailedToParseInput, err)
```

Comment:
Nothing actually returns an error. I put this in place a little speculatively, because if we ever evaluate expressions in variable definitions (see code comments / known limitations README in this PR), we'll have an error then, so it might make sense to front-load a decision about what type it should wrap.